### PR TITLE
Fixes #62. Closes #61.

### LINKED ISSUE #61: Bad email validation messages
When the view validates email addresses the message is bad. It says incorrect email when it probably should say "invalid email". Also, it doesn't allows caps in the email.

### LINKED ISSUE #62: Three way match happened
Log in with user 1 and select some restaurants.
Log out.
Log in with user 2 and select a different set of restaurants so as to not get matched
Open new incognito window.
Log in with user 3 in incognito. Select restuarants from both user 1 and user 2 choices.
Three way match.

screen shot 2015-03-30 at 12 45 29 am

### PR DESCRIPTION
**Pull request by paymog**
I decided against modifying the regex to allow caps. Don't
wanna mess things up

Fixes #62. Closes #61.

### CODE REVIEW

File: webapp/public/javascripts/controllers.js [hunk 1]
```diff
@@ -28,7 +29,7 @@ lunchrControllers.controller('MainPageController', ['$scope', '$http', '$state',
 
             if(/^[_a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/.test($scope.email) == false)
             {
-                $scope.errorMessages = "Invalid Email";
+                $scope.errorMessages = "Incorrect email";
                 return;
             }
```
Comment (kyleleung75) [inline webapp/public/javascripts/controllers.js]: Should we provide a better error message so the user knows not to use numbers or uppercase?

File: webapp/public/javascripts/controllers.js [hunk 2]
```diff
@@ -74,7 +75,7 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
 
             if(/^[_a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/.test($scope.email) == false)
             {
-                $scope.errorMessages = "Incorrect Email";
```
Comment (kyleleung75) [inline webapp/public/javascripts/controllers.js]: Same as comment above

File: webapp/public/javascripts/controllers.js (hunk 1 above)
Comment (paymog) [inline webapp/public/javascripts/controllers.js]: But it seems numbers are allowed. We should ask madeline exactly what isn't allowed.

Comment (kyleleung75) [inline webapp/public/javascripts/controllers.js]: Oh my bad, but yeah. I think a more meaningful error so they know how to input a valid email, at least on the registration page

Comment (MaddieLvrg) [inline webapp/public/javascripts/controllers.js]: Numbers and special characters should be allowed. The requirements are no uppercase letters and an @ symbol. If a period is included there must be following text.

**Comment by BrandonWade** [comment]
It seems odd to me to not allows uppercase characters for emails...Why not just allow them and set it to lowercase on the server?

**Comment by kjrcda** [comment]
K we can discuss  this in the room. This regex checks for a valid email address x@x.com. The server converts all emails to lower case regardless of what the user types.

**Comment by BrandonWade** [comment]
to not allow, rather

**Pull request merged**